src/doc/grammar.md

```diff
@@ -510,8 +510,8 @@ unit_expr : "()" ;
 ### Structure expressions
 
 ```antlr
-struct_expr : expr_path '{' ident ':' expr
-                      [ ',' ident ':' expr ] *
+struct_expr : expr_path '{' ident [ ':' expr ] ?
+                      [ ',' [ ident [ ':' expr ] ? ] *
```

Comment:
Braces are not balanced (and there’s the same mistake in the cleaner version as well). Extracting into `struct_expr_field_init` sounds like a good idea regardless. Makes it easier to read.